cishouseholds/pipeline/pipeline_stages.py

```diff
@@ -624,23 +625,51 @@ def lookup_based_editing(
         input file path name for travel_countries corrections lookup file
     edited_table
     """
-
     df = extract_from_table(input_table)
-    cohort_lookup = extract_from_table(cohort_lookup_table)
-    travel_countries_lookup = extract_from_table(travel_countries_lookup_table)
-    tenure_group = extract_from_table(tenure_group_table).select(
-        "UAC", "numAdult", "numChild", "dvhsize", "tenure_group"
-    )
-    for lookup_table_name, lookup_df, join_on_column_list in zip(
-        [cohort_lookup_table, travel_countries_lookup_table, tenure_group_table],
-        [cohort_lookup, travel_countries_lookup, tenure_group],
-        [["participant_id", "old_cohort"], ["been_outside_uk_last_country_old"], ["UAC"]],
-    ):
-        check_lookup_table_joined_columns_unique(
-            df=lookup_df, join_column_list=join_on_column_list, name_of_df=lookup_table_name
-        )
 
-    df = transform_from_lookups(df, cohort_lookup, travel_countries_lookup, tenure_group)
+    # COHORT
+    lookup_df_cohort = extract_from_table(cohort_lookup_table)
+    check_lookup_table_joined_columns_unique(
+        df=lookup_df_cohort, join_column_list=["participant_id", "old_cohort"], name_of_df=cohort_lookup_table
+    )
+    lookup_df_cohort = lookup_df_cohort.withColumnRenamed("old_cohort", "study_cohort")
+    df = null_safe_join(
+        left_df=df, right_df=lookup_df_cohort, null_safe_on=["participant_id", "study_cohort"], null_unsafe_on=[]
+    )
+    # TENURE
+    tenure_group_df = extract_from_table(tenure_group_table)
+    check_lookup_table_joined_columns_unique(
+        df=tenure_group_df,
+        join_column_list=["UAC"],
+        name_of_df=tenure_group_table,
+    )
+    for key, value in column_name_maps["tenure_group_variable_map"].items():
+        tenure_group_df = tenure_group_df.withColumnRenamed(key, value)
+    tenure_group_df = tenure_group_df.withColumnRenamed("UAC", "ons_household_id")
+
+    df = null_safe_join(
+        left_df=df,
+        right_df=tenure_group_df,
+        null_safe_on=["ons_household_id"],
+        null_unsafe_on=[],
+    )
+    # TRAVEL
+    lookup_df_travel = extract_from_table(travel_countries_lookup_table)
```

Comment:
The ticket asks for this logic to be wrapped into a unit level function, so it can be reused for travel and cohort. At the minute it's very similar code copied here and above